### ISSUE TITLE
Feature request: list predictions for NER

### ISSUE THREAD

**Issue by avant1** [ThilinaRajapakse/simpletransformers#54]
Hey there.
I've started using this package recently, and I think it is awesome! 

However I've noticed one possible issue with NER predictions format. It uses dict where keys are words itself. But same words (or tokens) within one text may have different labels depending on the context.

Consider the following example:
```
Bobby found 5 apples by 5 PM
```
Here first `5` token may have `N_FRUITS` label, second one may have `TIME_HOUR` tag. But predictions will have only one dict entry with key `5`.

I would propose output format that can be described as `List[Tuple[str, str]]`:
```
predictions = [
    ('Bobby', 'O'),
    ('found', 'O'),
    ('5', 'N_FRUITS'),
    ('apples', 'O'),
    ('by', 'O'),
    ('5', 'TIME_HOUR'),
    ('PM', ''),
]
```

It would be trivial to change current dict format into something like that, but we should take care about backward compatibility.

What do you think?

**Comment by ThilinaRajapakse** [comment]
Excellent point!

Building on your proposed format, I think we could use a list of dicts rather than a list of tuples so that it's closer to the original format.

The situation you described with duplicate words will probably be tough to debug if someone runs across this issue. So I think it would be better to enforce this change even though it breaks backward compatibility.

**Comment by avant1** [comment]
Dict based format looks good to me too.
Looks like you've already implemented this change in f60db784c. It works fine, thanks!

I think this issue can be closed.
